app/src/main/java/com/lastaoutdoor/lasta/data/db/ActivitiesRepositoryImpl.kt

```diff
@@ -0,0 +1,70 @@
+package com.lastaoutdoor.lasta.data.db
+
+import com.google.firebase.auth.FirebaseUser
+import com.google.firebase.firestore.FieldValue
+import com.google.firebase.firestore.FirebaseFirestore
+import com.google.firebase.firestore.SetOptions
+import com.lastaoutdoor.lasta.data.model.profile.ActivitiesDatabaseType
+import com.lastaoutdoor.lasta.repository.ActivitiesRepository
+import javax.inject.Inject
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.tasks.await
+
+private const val ACTIVITIES_DB_NAME = "activities_database"
+
+class ActivitiesRepositoryImpl @Inject constructor(private val database: FirebaseFirestore) :
+    ActivitiesRepository {
+  private val activityConverter = ActivityConverter()
+
+  private fun addUserToActivitiesDatabase(user: FirebaseUser) {
+    val userDocumentRef = database.collection(ACTIVITIES_DB_NAME).document(user.uid)
+    val userData =
+        hashMapOf(
+            "Hiking" to arrayListOf<ActivitiesDatabaseType.Trail>(),
+            "Climbing" to arrayListOf<ActivitiesDatabaseType.Climb>())
+    userDocumentRef.set(userData, SetOptions.merge())
+  }
+
+  @Suppress("UNCHECKED_CAST")
+  override suspend fun getUserActivities(
+      user: FirebaseUser,
+      activityType: ActivitiesDatabaseType.Sports
+  ): List<ActivitiesDatabaseType> {
+    val userDocumentRef = database.collection(ACTIVITIES_DB_NAME).document(user.uid)
+    val trailList: ArrayList<ActivitiesDatabaseType> = arrayListOf()
+    val document = userDocumentRef.get().await()
+
+    if (document != null) {
+      val activityArray = document.get(activityType.toString()) as? List<*>
+      if (activityArray != null) {
+        for (item in activityArray) {
+          trailList.add(
+              activityConverter.databaseToActivity(item as HashMap<String, Any>, activityType))
+        }
+      } else {
+        println("No 'Hiking' array found or it's not an array")
```

Comment:
Are errors displayed on the app with `println()` or just on the dev console ? Maybe it would be coherent to signal the user of an error occuring

Reply:
Thank you for the feedback! It was just a default error handling code that was not yet testes. It would indeed print in the console but that is not meaningful for the application, so I changed it to return simply an empty list